### PR TITLE
[core] fix skew bug of stream read unware bucket table

### PR DESCRIPTION
**Pull request by wwj6591812**
Now when stream read a unware bucket table with comsume_id and consumer.mode = exactly-once. SKEW will happy, this is a bug, now fix it.
image

### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java [hunk 1]
```diff
@@ -230,23 +234,52 @@ public static DataStream<RowData> buildSource(
...
-                                return ChannelComputer.select(key.f0, key.f1, numPartitions);
-                            }
-                            return ChannelComputer.select(key.f1, numPartitions);
-                        },
-                        split -> {
-                            DataSplit dataSplit = (DataSplit) split;
-                            return Tuple2.of(dataSplit.partition(), dataSplit.bucket());
-                        })
-                .transform(name + "-Reader", typeInfo, new ReadOperator(readBuilder));
+        if (shuffleByPartition) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java:259]: You really want this? The `streaming-read.shuffle-by-partition` is true by default. It means?

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/AppendOnlyTableITCase.java [hunk 2]
```diff
@@ -85,6 +85,22 @@ public void testReadWrite() {
         assertThat(rows).containsExactlyInAnyOrder(Row.of("AAA"), Row.of("BBB"));
     }
 
+    @Test
+    public void testReadWriteWithUnwareBucket() throws Exception {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/AppendOnlyTableITCase.java:89]: What is this test for?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java (hunk 1 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java:259]: In my business scenario, I will set `streaming-read.shuffle-by-partition` to false.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/AppendOnlyTableITCase.java (hunk 2 above)
Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/AppendOnlyTableITCase.java]: Sorry, I have change the method name to testReadUnwareBucketTableWithRebalanceShuffle.

This test is to test that the data read under the following conditions is correct：
1、consumer.mode=exactly-once
2、Streaming read unware bucket table.
3、Send Split from MonitorFunction to ReadOperator is rebalance.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java:259]: This is bug, you should fix it.

Comment (wwj6591812) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/MonitorFunction.java]: done

**Review by JingsongLi — APPROVED** [review]
Thanks @wwj6591812 , looks good to me!

**Pull request merged**